src/server/pps/server/api_server.go

```diff
@@ -2313,6 +2225,18 @@ func setInputDefaults(pipelineName string, input *pps.Input) {
 	})
 }
 
+func (a *apiServer) stopAllJobsInPipeline(txnCtx *txncontext.TransactionContext, pipeline *pps.Pipeline) error {
+	// Using ReadWrite here may load a large number of jobs inline in the
```

Comment:
This seems like something that should be handled by the pipeline controller (I think it already does in certain cases). Including all of the jobs in this transaction seems problematic for fast processing jobs, and I don't think it is necessary to stop all of the jobs in a single transaction. I may be missing something, but the way I am thinking about it is that we have one transaction that handles the state transition for the pipeline info, then the pipe controller sees that state transition and spins down the workers and stops the jobs, then spins up new workers.

Reply:
This is what the 'pipeline finisher' used to do at the end of transactions in PFS and I agree it's not great, but it's better than it was before.  Before, it would query all of the output commits in the output repo (_outside of the transaction_), then attempt to stop the job associated with each output commit inside the transaction (leading to false positives and possible race conditions).  I'm up for rethinking this, but I'd prefer to tackle that separately.

Reply:
Yeah, this is one that we should get a handle on before 2.0 GA, but not this PR. I'll add a card.